crates/rpc/src/lib.rs

```diff
@@ -1,23 +1,63 @@
+#![allow(unused_variables)]
+#![allow(clippy::needless_borrow)]
 use std::sync::Arc;
 
+use actix_web::{App, HttpServer, dev::ServerHandle, middleware, web::Data};
 use config::ServerConfig;
 use ream_network_spec::networks::NetworkSpec;
 use ream_storage::db::ReamDB;
-use routes::get_routes;
 use tracing::info;
-use utils::error::handle_rejection;
-use warp::{Filter, serve};
+
+use crate::routes::register_routers;
 
 pub mod config;
 pub mod handlers;
 pub mod routes;
 pub mod types;
-pub mod utils;
 
 /// Start the Beacon API server.
-pub async fn start_server(network_spec: Arc<NetworkSpec>, server_config: ServerConfig, db: ReamDB) {
-    let routes = get_routes(network_spec, db).recover(handle_rejection);
+pub async fn start_server(
+    server_config: ServerConfig,
+    network_spec: Arc<NetworkSpec>,
+    db: ReamDB,
+) -> std::io::Result<()> {
+    info!(
+        "starting HTTP server on {:?}",
+        server_config.http_socket_address
+    );
+    // create the stop handle container
+    let stop_handle = Data::new(StopHandle::default());
+
+    let server = HttpServer::new(move || {
+        let stop_handle = stop_handle.clone();
+        App::new()
+            .wrap(middleware::Logger::default())
+            .app_data(stop_handle)
+            .app_data(Data::from(network_spec.clone()))
+            .app_data(Data::new(db.clone()))
+            .configure(register_routers)
```

Comment:
```suggestion
    let server = HttpServer::new(move || {
        let stop_handle = stop_handle.clone();
        App::new()
            .wrap(middleware::Logger::default())
            .app_data(stop_handle)
            .app_data(Data::from(network_spec))
            .app_data(Data::new(db))
            .configure(register_routers)
```

Could we remove the clones?